### PR TITLE
Fix deprecated functions in ONNX and boost + type conversion for ConfigurePythonTest

### LINKED ISSUE #1208: Compiler warnings
**Describe the bug**
In the idea of having clean compiler logs, here are a few more warnings I see:

Type "control reaches end of non-void function"
```
/Users/tav/Documents/1Research/LDMX/ldmx-sw/DetDescr/src/DetDescr/HcalGeometry.cxx: In member function 'std::vector<double> ldmx::HcalGeometry::rotateGlobalToLocalBarPosition(const std::vector<double>&, const ldmx::HcalID&) const':
/Users/tav/Documents/1Research/LDMX/ldmx-sw/DetDescr/src/DetDescr/HcalGeometry.cxx:77:1: warning: control reaches end of non-void function [-Wreturn-type]
   77 | }
      | ^
/Users/tav/Documents/1Research/LDMX/ldmx-sw/DetDescr/src/DetDescr/HcalGeometry.cxx: In member function 'ldmx::HcalGeometry::ScintillatorOrientation ldmx::HcalGeometry::getScintillatorOrientation(ldmx::HcalID) const':
/Users/tav/Documents/1Research/LDMX/ldmx-sw/DetDescr/src/DetDescr/HcalGeometry.cxx:128:1: warning: control reaches end of non-void function [-Wreturn-type]
  128 | }
      | ^
```

Type " declaration of 'm' shadows a global declaration "
```
/Users/tav/Documents/1Research/LDMX/ldmx-sw/SimCore/G4DarkBreM/app/simulate.cxx: In constructor 'g4db::example::Hunk::Hunk(double, const string&)':
/Users/tav/Documents/1Research/LDMX/ldmx-sw/SimCore/G4DarkBreM/app/simulate.cxx:121:37: warning: declaration of 'm' shadows a global declaration [-Wshadow]
  121 |   Hunk(double d, const std::string& m)
      |                  ~~~~~~~~~~~~~~~~~~~^
In file included from /usr/local/include/Geant4/CLHEP/Units/PhysicalConstants.h:42,
                 from /usr/local/include/Geant4/G4ParticleDefinition.hh:58,
                 from /usr/local/include/Geant4/G4ParticleTable.hh:58,
                 from /usr/local/include/Geant4/G4VUserPhysicsList.hh:92,
                 from /usr/local/include/Geant4/G4VModularPhysicsList.hh:59,
                 from /usr/local/include/Geant4/QBBC.hh:42,
                 from /Users/tav/Documents/1Research/LDMX/ldmx-sw/SimCore/G4DarkBreM/app/simulate.cxx:11:
/usr/local/include/Geant4/CLHEP/Units/SystemOfUnits.h:108:23: note: shadowed declaration is here
  108 |   static const double m  = meter;
      |                       ^
```

Type "deprecated"
```
/Users/tav/Documents/1Research/LDMX/ldmx-sw/Tools/src/Tools/ONNXRuntime.cxx: In constructor 'ldmx::Ort::ONNXRuntime::ONNXRuntime(const string&, const Ort::SessionOptions*)':
/Users/tav/Documents/1Research/LDMX/ldmx-sw/Tools/src/Tools/ONNXRuntime.cxx:70:30: warning: 'void Ort::detail::TensorTypeAndShapeInfoImpl<T>::GetDimensions(int64_t*, size_t) const [with T = Ort::detail::Unowned<const OrtTensorTypeAndShapeInfo>; int64_t = long int; size_t = long unsigned int]' is deprecated: use GetShape() [-Wdeprecated-declarations]
   70 |     tensor_info.GetDimensions(input_node_dims_[input_name].data(), num_dims);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /Users/tav/Documents/1Research/LDMX/ldmx-sw/Tools/include/Tools/ONNXRuntime.h:10,
                 from /Users/tav/Documents/1Research/LDMX/ldmx-sw/Tools/src/Tools/ONNXRuntime.cxx:2:
/usr/local/include/onnxruntime_cxx_api.h:858:41: note: declared here
  858 |   [[deprecated("use GetShape()")]] void GetDimensions(int64_t* values, size_t values_count) const;  ///< Wraps OrtApi::GetDimensions
      |                                         ^~~~~~~~~~~~~
/Users/tav/Documents/1Research/LDMX/ldmx-sw/Tools/src/Tools/ONNXRuntime.cxx:92:30: warning: 'void Ort::detail::TensorTypeAndShapeInfoImpl<T>::GetDimensions(int64_t*, size_t) const [with T = Ort::detail::Unowned<const OrtTensorTypeAndShapeInfo>; int64_t = long int; size_t = long unsigned int]' is deprecated: use GetShape() [-Wdeprecated-declarations]
   92 |     tensor_info.GetDimensions(output_node_dims_[output_name].data(), num_dims);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/local/include/onnxruntime_cxx_api.h:2088,
                 from /Users/tav/Documents/1Research/LDMX/ldmx-sw/Tools/include/Tools/ONNXRuntime.h:10,
                 from /Users/tav/Documents/1Research/LDMX/ldmx-sw/Tools/src/Tools/ONNXRuntime.cxx:2:
/usr/local/include/onnxruntime_cxx_inline.h:1061:13: note: declared here
 1061 | inline void TensorTypeAndShapeInfoImpl<T>::GetDimensions(int64_t* values, size_t values_count) const {
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Boost deprecation
```
In file included from /usr/include/boost/smart_ptr/detail/sp_thread_sleep.hpp:22,
                 from /usr/include/boost/smart_ptr/detail/yield_k.hpp:23,
                 from /usr/include/boost/smart_ptr/detail/spinlock_gcc_atomic.hpp:14,
                 from /usr/include/boost/smart_ptr/detail/spinlock.hpp:42,
                 from /usr/include/boost/smart_ptr/detail/spinlock_pool.hpp:25,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:29,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /usr/include/boost/python/converter/shared_ptr_to_python.hpp:12,
                 from /usr/include/boost/python/converter/arg_to_python.hpp:15,
                 from /usr/include/boost/python/call.hpp:15,
                 from /usr/include/boost/python/object_core.hpp:14,
                 from /usr/include/boost/python/args.hpp:22,
                 from /usr/include/boost/python.hpp:11,
                 from /Users/tav/Documents/1Research/LDMX/BugFixes/ldmx-sw/DetDescr/src/DetDescr/DetectorIDBindings.cxx:2:
/usr/include/boost/bind.hpp:36:1: note: '#pragma message: The practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is deprecated. Please use <boost/bind/bind.hpp> + using namespace boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior.'
   36 | BOOST_PRAGMA_MESSAGE(
      | ^~~~~~~~~~~~~~~~~~~~
/usr/include/boost/detail/iterator.hpp:13:1: note: '#pragma message: This header is deprecated. Use <iterator> instead.'
   13 | BOOST_HEADER_DEPRECATED("<iterator>")
      | ^~~~~~~~~~~~~~~~~~~~~~~
```

and

```
  /home/runner/work/ldmx-sw/ldmx-sw/Framework/test/ConfigurePythonTest.cxx: In function 'void CATCH2_INTERNAL_TEST_0()':
  /home/runner/work/ldmx-sw/ldmx-sw/Framework/test/ConfigurePythonTest.cxx:166:15: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
    166 |     args[0] = "9000";
```

**Additional context**

Similar issue at https://github.com/LDMX-Software/ldmx-sw/issues/1199

### PR DESCRIPTION
**Pull request by tvami**
I am updating _ldmx-sw_, here are the details.

### What are the issues that this addresses?
This resolves https://github.com/LDMX-Software/ldmx-sw/issues/1208

## Check List
- [x] I successfully compiled _ldmx-sw_ with my developments
- [x] I ran my developments and the following shows that they are successful.

### CODE REVIEW

**Comment by tvami** [comment]
@tomeichlersmith what do you think about what I did here for boost? This (partially) solves the warning (the `<iterator>` part I cannot do anything about), but I think it's also the reason for the change in the logs. I could remove that from this PR and have you bump up the boost version and come back to this later. Or keep it as is and revisit this warning whenever the boost update happens.

**Review by tomeichlersmith — COMMENTED** [review]
I think this makes sense. The deprecation methods within Boost are from the Boost.Python library which we use for the bindings within DetDescr. I'm not super worried about the text changes to the log [0] since the changes are only on a handful of events out of 10k. Maybe the change to Boost is somehow affecting the TDecompSVD class in ROOT? It seems to be an edge case at least.

All I'm worried about is the ONNX change since I'm so unfamiliar with it, but a few test-runs with someone more familiar with the BDT would be able to confirm that GetShape and GetDimension are equivalent for us.

<details>

  <summary> [0] Changes to logs were in the signal validation sample when PR was originally opened.</summary>

```
  <  [ ecalVeto ] 0 :   MIP tracking completed; found 4 straight tracks and 0 lin-reg tracks
  ---
  >  [ ecalVeto ] 0 :   MIP tracking completed; found 4 straight tracks and 2 lin-reg tracks
  13163c13163
  <  [ ecalVeto ] 0 :   MIP tracking completed; found 0 straight tracks and 0 lin-reg tracks
  ---
  >  [ ecalVeto ] 0 :   MIP tracking completed; found 0 straight tracks and 1 lin-reg tracks
  73962c73962
  <  [ ecalVeto ] 0 :   MIP tracking completed; found 2 straight tracks and 0 lin-reg tracks
  ---
  >  [ ecalVeto ] 0 :   MIP tracking completed; found 2 straight tracks and 1 lin-reg tracks
  101379c101379
  <  [ ecalVeto ] 0 :   MIP tracking completed; found 3 straight tracks and 0 lin-reg tracks
  ---
  >  [ ecalVeto ] 0 :   MIP tracking completed; found 3 straight tracks and 1 lin-reg tracks
```

</details>

File: DetDescr/src/DetDescr/DetectorIDBindings.cxx [hunk 1]
```diff
@@ -1,4 +1,6 @@
 #if DETECTORID_BINDINGS_ENABLED
+// This we may be able to remove after boost update
+#define BOOST_BIND_GLOBAL_PLACEHOLDERS
 #include <boost/python.hpp>
```
Comment (tomeichlersmith) [inline DetDescr/src/DetDescr/DetectorIDBindings.cxx:4]: The more I think about it, the less I'm convinced this is causing the change in the logs.

This `#define` is within a source file and so would not affect how the Ecal library is compiled. I guess this define would be applied (?) after the Ecal library is linked to the DetDescr library, but I have not heard of `#define` being used in this way.

Comment (tvami) [reply DetDescr/src/DetDescr/DetectorIDBindings.cxx:4]: Yes I guess you are right it shouldnt be connected

**Comment by tomeichlersmith** [comment]
Alright, the change in lin-reg tracks within the ECal Veto is still persisting. Here is a quick summary,

Sample | Change
---|---
it_pileup | 4 Events: 2 -> 0; 43 Events: 1 -> 0
reduced_ldmx | 1 Event: 0 -> 1
inclusive | Does not run EcalVeto
hcal | Does not run EcalVeto
ecal_pn | No Change (probably due to change in log level) [^1]
signal | 1 Event: 0 -> 1; 3 Events 0 -> 2

While I am mildly curious about the cause behind this, I think it could reasonably be from a change in the underlying runner's software (OS or version of docker for example) that causes a slight change for us.

[^1]: The ecal_pn fails validation since the logs change by a few characters when reporting the timing of different Tracking components at the end of processing.

**Comment by tvami** [comment]
> No MIP Tracking Change

I'm not sure that's just not hidden given that the log level is different from the rest of them :/

**Comment by tvami** [comment]
> OS or version of docker for example

Where did you do this? This https://github.com/LDMX-Software/docker/pull/97 you closed, right?

**Comment by tomeichlersmith** [comment]
No I mean the OS of the runner launched by GitHub. I'd be surprised if that affects things, but throwing stuff out there.

You're right about the no change probably being from the log level.

**Comment by tomeichlersmith** [comment]
I tried running locally and the log diff showed even more differences. I need to check my local configuration and stuff but I fear that this is not something that is easy to reproduce and test.

**Comment by tomeichlersmith** [comment]
@tvami I see two ways forward:

1. Just Merge -- I think this isn't too big of a deal and encourages us to double check the MIP track counts (which we should probably do anyways).
2. Separate Changes into Different PRs -- This would then allow us to see if there is a single change that is causing the change in MIP track counts and then we could go from there.

What do you think?

**Comment by tvami** [comment]
I still havent had the chance to look at the BDT score and other plots in higher stats. It's not a lot of work, I just keep getting distracted with other things :D But yeah let me post some plots by tomorrow. If you want, you can approve based on the code changes, and then I'll post the plots and we can merge later if they look good

**Comment by tvami** [comment]
CompareBDT

The BDT values are exactly the same for ECAL pn, which makes me happy with the changes.
I also looked at the nStraightTracks and the nLigRegTracks, they also look identical.

**Review by tomeichlersmith — APPROVED** [review]
Sweet, thanks for doing that double check. I feel mich more confident that the log changes are neglible.
